### PR TITLE
Display branches in the tree in the planned order; update leaf colors to separate similar shades

### PR DESCRIPTION
**Pull request by rlskoeser**
in this pr:
- updates the tree so leaves are organized by branch in planned sequence from left to right
- revises colors so that similar colors are further apart
- place labels directly on top of the leaves instead of treating them as separate nodes within the network

This layout still needs some more refining (some leaves are going off the edges), but I think it's a lot more successful and we're heading in the right direction. In particular, the branches and centuries are much more accurate right now.

implementation details:
- I added a x-coordinate force to draw leaves to their branch location; right now I'm using a vertical line with an increasing force based on century (earlier leaves are less strongly drawn); we could revise to an angled line if necessary
- I'm drawing vertical lines on the debug mode so that you can see where the center of each branch is
- labels are now placed directly over the leaves, per Gissoo's suggestion; this simplified the code substantially

## testing notes 
@jhimpele please review on render site for this pr https://lenape-timetree-dev-pr-123.onrender.com/
- [x] confirm that leaves are more clearly separated by branch
- [x] confirm that color separation is better with this branch and color order
- [x] confirm that branch order follows the desired sequence left to right

### CODE REVIEW

**Comment by jhimpele** [comment]
The original had a different sequence of branches...Please switch the university and resistance and resurgence sequence. But when we meet, we may want to consider the overall shape, which is less-tree like and flatter now. Switching the university and RR makes the tree flatter in the middle, where one would expect it to be tallest!

(Also, there are ghost labels visible. Is that expected?)

**Comment by jhimpele** [comment]
Here's a screenshot of https://lenape-timetree-dev-pr-123.onrender.com
on Safari....
Screenshot 2023-03-27 at 10 37 53 AM

File: assets/js/timetree.js [hunk 1]
```diff
@@ -326,15 +331,25 @@ function TreeGraph({ nodes, links, centuries }) {
...
         .forceY()
         .y((node) => centuryY(node))
         .strength(forceStrength.centuryY)
+    )
+    .force(
+      "x",
+      d3
+        .forceX()
+        .x((node) => branchX(node))
+        .strength((node) => {
```
Comment (quadrismegistus) [inline assets/js/timetree.js:346]: So this is a wholly new x-axis force for the branches.

**Review by quadrismegistus — APPROVED** [review]
Looks great. Very clever uses of the d3 forces

**Comment by gissoo** [comment]
@rlskoeser this looks great!! Everything is so much more readable (except the problem Jeff has mentioned on Safari)

File: assets/js/timetree.js [hunk 2]
```diff
@@ -97,16 +101,30 @@ sortedLeaves.forEach((leaf) => {
...
-    acc[b] = [];
+  // check that branch is in our list
+  if (b in branches) {
+    if (acc[b] == undefined) {
+      acc[b] = [];
+    }
+    acc[b].push(leaf);
+  } else {
+    // report unknown branchand omit from  the tree
+    console.log(`Unknown branch: ${b}`);
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply assets/js/timetree.js:124]: It hasn't happened yet, but it's user-editable content that we're not checking anywhere so I worry about content getting lost due to a typo. This probably isn't that helpful of a solution, but I wasn't sure where to check it properly.  Maybe could figure out a PR check that would flag it?

File: assets/js/timetree.js (hunk 1 above)
Comment (rlskoeser) [reply assets/js/timetree.js:346]: yes

File: assets/js/timetree.js [hunk 3]
```diff
@@ -326,15 +331,25 @@ function TreeGraph({ nodes, links, centuries }) {
...
+    )
+    .force(
+      "x",
+      d3
+        .forceX()
+        .x((node) => branchX(node))
+        .strength((node) => {
+          if (node.century != undefined) {
+            // apply the force more strongly the further up the tree we go
+            return forceStrength.branchX * (node.century - 14);
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply assets/js/timetree.js:349]: This is so the leaves in the lower part of the tree (earlier centuries) aren't drawn as strongly to the x coordinate for their branch. (Maybe that's a different way to say what you asked?)

**Comment by rlskoeser** [comment]
updated as follows:
- revised branch order agreed on in the meeting
- update branch color order to match the new sequence
- rename "Lands and Waters" branch to "Lands + Waters"
- move 2006 Lawsuit leaf from Lands + Waters to Resistance + Resurgence
- fix leaf text shadow so it is not broken on Safari (there's a Safari bug, doesn't support multiple text shadows)
- adjust size of leaf glow effect

@gissoo a few items:
- I think I got all the colors switched, but would appreciate a quick check through the various states to confirm
- Safari has a bug, it only supports one text shadow, so I revised what I had (zero offset, larger blur); let me know if you want me to make any adjustments
- The leaf glow effect was for your 40x width but we resized to half that for implementation, so I just halved all of the numbers on the drop-shadow effect  - does that sound right? does it look ok?

File: assets/js/timetree.js [hunk 4]
```diff
@@ -167,45 +185,24 @@ for (const branch in leavesByBranch) {
...
 // branch style color sequence; set class name and control with css
-let branchStyles = ["a", "b", "c", "d", "e"];
-
 function getBranchStyle(branchName) {
-  let branchIndex = Object.keys(leavesByBranch).indexOf(branchName);
-  // let branchIndex = Object.keys(branches).indexOf(branchName);
-  let branchStyle = branchStyles[branchIndex];
-  if (branchStyle != undefined) {
-    return "branch-" + branchStyles[branchIndex];
+  let branchSlug = branches[branchName];
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply assets/js/timetree.js:203]: I'm not sure I understand the question, sorry! 

I'm generating my own slugs to use in the js code and the css; not sure how to manage better, open to suggestions.

**Pull request merged**